src/main/java/org/folio/processing/mapping/mapper/writer/marc/MarcRecordModifier.java

```diff
@@ -684,6 +694,18 @@ private void replaceDataField(DataField fieldReplacement, String fieldTag, char
     }
   }
 
+  boolean isNonRepeatableField(DataField field) {
+    // is any of 1xx fields
+    if (field.getTag().compareTo(TAG_100) > -1 && field.getTag().compareTo(TAG_199) < 1) {
```

Comment:
can `filed.getTag()` be null?

Reply:
If it can be null, I think you should switch the compare/equality around. `TAG_100 == field.getTag()` instead of `field.getTag() == TAG_100`